### DISCUSSION TITLE
Fixing bugs in the main branch

### DISCUSSION THREAD

**Discussion by DerrickLau** [RT-Thread/rt-thread#7915] [Q&A] [answered]
Greetings all:

I recently downloaded the main/master branch of rt-thread and attempted to use the rt-thread-env toolset and scons to build the raspberry pi 4 project.  It wouldn't build until I modified some code.  Now that I have made changes to ensure the project builds, I'd like to inquire what is the process to merge a code change into the main branch, as I assume I cannot check out a file in the main branch.  Also, who coordinates the checkouts, checkins and merges of code?

Please advise,

Thanks,
Derrick

**Comment by supperthomas** [comment, marked as answer]
please follow this link for PR.
https://github.com/RT-Thread/rt-thread/blob/83a2863ab6d09ccb440e819ec6fc9d4f491cca8c/.github/CONTRIBUTING.md
